packages/shared/src/features/onboarding/hooks/useFunnelNavigation.ts

```diff
@@ -0,0 +1,177 @@
+import { useMemo, useState, useEffect, useCallback } from 'react';
```

Comment:
We need some more comments here to make it maintainable

Reply:
Updated ✔️